src/main/java/com/savvato/tribeapp/controllers/AttributesAPIController.java

```diff
@@ -45,7 +45,7 @@ public ResponseEntity<Boolean> applyPhraseToUser(@RequestBody @Valid AttributesR
             return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(null);
 
         if (phraseService.isPhraseValid(req.adverb, req.verb, req.preposition, req.noun)) {
-            phraseService.applyPhraseToUser(req.id, req.adverb, req.verb, req.preposition, req.noun);
+            phraseService.applyPhraseToUser(req.userId, req.adverb, req.verb, req.preposition, req.noun);
```

Comment:
@haxwell when I updated the req.id in this call to req.userId, it did change the AttributeRequestDTO. I noticed that the AttributeRequestDTO is actually specified in the endpoint signature:
`@RequestMapping(value = { "/api/attributes" }, method=RequestMethod.POST) public ResponseEntity<Boolean> applyPhraseToUser(@RequestBody @Valid AttributesRequest req) {`
This is contrary to what we discussed in the last meeting, where you said that AttributeRequstDTO should be for responses and the id field is for the attribute id. I think this may have been overlooked when I worked on separating "attributes" and "phrase" functionality in TRIB-33. How do you want this to work? Any changes, or should I leave it as is?

Reply:
@mrsbluerose The *Request objects describe the possible fields that the JSON body of a request to an associated endpoint might contain.

So AttributesRequest's id field would be the ID of the attribute the client was concerned with.

The response objects end in DTO. They describe the data returned in response to an endpoint call.

So, I think the way to resolve this is to add a field to the AttributesRequest for 'userId'. Keep the 'id' field, and add a 'userId' field.